src/main/scala/optics/poly/Example.scala

```diff
@@ -0,0 +1,33 @@
+package optics.poly
+
+import optics.poly.dsl._
+
+case class Fub(bab: Int)
+case class Bar(fub: Fub)
+case class Foo(bar: Option[Bar])
+
+object elsewhere {
+  given [A] as Iso[Option[A], Either[Unit, A]] =
+    Iso(
+      o => o.map(Right(_)).getOrElse(Left(())),
+      {
+        case Left(_) => None
+        case Right(a) => Some(a)
+      }
+    )
+}
+
+object Foo {
+
+  import elsewhere.{given _}
+
+  val llens = Focus[Foo](_.bar.?.fub)
```

Comment:
This isn't quite complete yet but this is actually returning an `Either[Unit, Bar]` at the moment!